modules/client/shared/src/main/scala/lucuma/itc/client/ItcResult.scala

```diff
@@ -18,119 +18,51 @@ import java.math.MathContext
 
 sealed trait ItcResult
 
-object ItcResult {
-
-  given Decoder[ItcResult] with
-    def apply(c: HCursor): Decoder.Result[ItcResult] =
-      c.downField("__typename").as[String].flatMap {
-        case "IntegrationTime"  => c.as[IntegrationTime].widen[ItcResult]
-        case "SourceTooBright"  => c.as[SourceTooBright].widen[ItcResult]
-        case "CalculationError" => c.as[CalculationError].widen[ItcResult]
-        case rt                 =>
-          DecodingFailure(s"Couldn't parse ItcResult as success or error: $rt", c.history).asLeft
-      }
-
-  given Eq[ItcResult] with
-    def eqv(x: ItcResult, y: ItcResult): Boolean =
-      (x, y) match {
-        case (s0: IntegrationTime, s1: IntegrationTime)   => s0 === s1
-        case (e0: SourceTooBright, e1: SourceTooBright)   => e0 === e1
-        case (e0: CalculationError, e1: CalculationError) => e0 === e1
-        case _                                            => false
-      }
-
-  final case class CalculationError(
-    originalMessages: List[String]
-  ) extends ItcResult {
-    val message: String = originalMessages.mkString(", ")
-  }
-
-  object CalculationError {
-
-    given Decoder[CalculationError] with
-      def apply(c: HCursor): Decoder.Result[CalculationError] =
-        c.downField("originalMessages").as[List[String]].map(CalculationError(_))
-
-    given Eq[CalculationError] =
-      Eq.by(_.originalMessages)
-
-  }
-
-  final case class SourceTooBright(
-    halfWellTime: BigDecimal
-  ) extends ItcResult {
-    val message: String = s"Source too bright, it saturates at $halfWellTime"
-  }
-
-  object SourceTooBright {
-
-    given Decoder[SourceTooBright] with
-      def apply(c: HCursor): Decoder.Result[SourceTooBright] =
-        c.downField("halfWellTime")
-          .as[BigDecimal]
-          .map(SourceTooBright(_))
-
-    given Eq[SourceTooBright] =
-      Eq.by(_.halfWellTime)
-
-  }
-  final case class IntegrationTime(
-    exposureTime:  TimeSpan,
-    exposures:     NonNegInt,
-    signalToNoise: SignalToNoise
-  ) extends ItcResult {
-
-    def stepSignalToNoise: Option[SignalToNoise] =
-      SignalToNoise.FromBigDecimalRounding.getOption(
-        BigDecimal(
-          (signalToNoise.toBigDecimal * signalToNoise.toBigDecimal / exposures.value)
-            .underlying()
-            .sqrt(MathContext.DECIMAL128)
-        )
+final case class IntegrationTime(
+  exposureTime:  TimeSpan,
+  exposures:     NonNegInt,
+  signalToNoise: SignalToNoise
+) extends ItcResult {
+
+  def stepSignalToNoise: Option[SignalToNoise] =
```

Comment:
Hmm, should `exposures` be `PosInt`?  We'll get a divide by 0 error otherwise if 0.  Or maybe we return `None` if exposures is 0?

Maybe we should add this to `SignalToNoise` while we're at it?  In an `extension` I mean.  What do you think?

```
def stepSignalToNoise(exposures: NonNegInt): Option[SignalToNoise]
```

Reply:
Sure, exposures has to be 1 or higher right. I'm checking the SN 0 case and returning an error if that happens

Reply:
I realized we can reuse `IntegrationTime` on the server and client where the only difference is we use `PosInt` in one place

Reply:
> Sure, exposures has to be 1 or higher

Won't this throw an exception if it is called with 0 exposures?

Reply:
If it is a PosInt it can't be zero right?

Reply:
Yes, actually we can unify the model for client and server and use PosInt